### DISCUSSION TITLE
finding the argocd server which manages a certain resource

### DISCUSSION THREAD

**Discussion by eranreshef-starkware** [argoproj/argo-cd#19531] [Q&A]
Is there a way for me to find out which argocd instance is responsible for a certain resource in a certain k8s repo?
Lets say I have 3 different argocd instances A, B, and C which control 3 different target clusters X, Y, and Z. I am now trying to debug a certain k8s resource in cluster Y and I would like to know which one of my argocd instances manages that cluster.
Is that information saved somewhere in the target clusters?

Thanks :pray:

**Comment by ChristianCiach** [comment]
The answer is probably "no". The next best thing you could do for the future is to use different tracking labels for different Argo CD instances.

For example, you could put this into your `argocd-cm` ConfigMap of your `A` instance:

```
application.instanceLabelKey: 'argocd.argoproj.io/instance-a'
```

It is probably not a good idea to change this for existing clusters...

Of course, this only works if you didn't change `application.resourceTrackingMethod` to `annotation`.

**Reply by ChristianCiach** [reply]
A nicer solution would be something like this:

* https://github.com/argoproj/argo-cd/pull/14945

But in the current state of this PR, the value `argocd` is unfortunately hardcoded.

**Comment by ChristianCiach** [comment]
With ArgoCD 2.13 you can now configure an `installationID` that will be added to all managed resources as an annotation named `argocd.argoproj.io/installation-id`.

https://argo-cd.readthedocs.io/en/release-2.13/user-guide/resource_tracking/#installation-id

**Comment by andrii-korotkov-verkada** [comment]
I think there would be a label like `argocd.argoproj.io/instance` or `app.kubernetes.io/instance` https://argo-cd.readthedocs.io/en/latest/user-guide/annotations-and-labels/#labels
